src/cpp/src/visual_language/qwen2vl/classes.cpp

```diff
@@ -319,6 +320,29 @@ ov::Tensor get_attention_mask(const std::vector<std::array<size_t, 3>>& reordere
     return attention_mask;
 }
 
+ov::Tensor get_cu_seqlens(const std::vector<std::array<size_t, 3>>& reordered_images_grid_thw) {
+    // Calculate cumulative sequence lengths for attention mask
+    std::vector<int32_t> cu_seqlens;
+    cu_seqlens.push_back(0);
+    int32_t cumsum = 0;
+    for (const auto& grid_thw : reordered_images_grid_thw) {
+        size_t slice_len = grid_thw.at(1) * grid_thw.at(2);
+        for (size_t t = 0; t < grid_thw.at(0); ++t) {
+            cumsum += slice_len;
+            cu_seqlens.push_back(cumsum);
+        }
+    }
+
+    ov::Tensor t_cu_seqlens = ov::Tensor(ov::element::i32, {cu_seqlens.size()});
+    auto* ptr = static_cast<int32_t*>(t_cu_seqlens.data());
+    for (size_t n = 0; n < cu_seqlens.size(); n++) {
+        ptr[n] = cu_seqlens[n];
+    }
```

Comment:
I think we can create OV sub-graph for computation of `cu_seqlens` from `reordered_images_grid_thw` and attach to the main graph. I don't want us to have more computation in C++, rather to offload this work to plugin side.

Reply:
hmm, sounds good.

Reply:
Not sure in memory increase. If you have it, then you have it in your C++ implementation too.
Please elaborate concern.

Reply:
Still not resolved?

Reply:
you don't need to compute attention_mask. You should compute `cu_seqlens` directly from `reordered_images_grid_thw`. 
For this you need to create ov::Model, compile it and execute in OV runtime.
You can also stitch this graph to the main one.

Reply:
CVS-170675 created to follow up this issue.